### PR TITLE
refresh bearer token from file to avoid stale tokens when using mount…

### PR DESCRIPTION
**Pull request by grosser**
…ed service account token

fixes https://github.com/abonas/kubeclient/issues/530

### CODE REVIEW

File: lib/kubeclient.rb [hunk 1]
```diff
@@ -358,6 +357,12 @@ def create_faraday_client(url = nil)
         if @auth_options[:username] && @auth_options[:password]
           connection.basic_auth(@auth_options[:username], @auth_options[:password])
         end
+
+        refresh_bearer_token_from_file
+        if auth = @headers[:Authorization]
+          req.headers['Authorization'] = auth
```
Comment (grosser) [inline lib/kubeclient.rb:363]: this is kinda sus ... did not see where this was supposed to be coming from before

File: lib/kubeclient.rb [hunk 2]
```diff
@@ -695,11 +698,9 @@ def validate_auth_options(opts)
...
 
     def validate_bearer_token_file
-      msg = "Token file #{@auth_options[:bearer_token_file]} does not exist"
-      raise ArgumentError, msg unless File.file?(@auth_options[:bearer_token_file])
-
-      msg = "Cannot read token file #{@auth_options[:bearer_token_file]}"
-      raise ArgumentError, msg unless File.readable?(@auth_options[:bearer_token_file])
+      file = @auth_options[:bearer_token_file]
+      raise ArgumentError, "Token file #{file} does not exist" unless File.file?(file)
+      raise ArgumentError, "Token file #{file} cannot be read" unless File.readable?(file)
```
Comment (grosser) [inline lib/kubeclient.rb:703]: just cleanup

**Comment by grosser** [comment]
with rebase this is now super clean :D

**Comment by kjetijor** [comment]
`@client = Kubeclient::Client.new(....)` going straight to `@client.api_valid?` seems to not work with this, it'll end up missing any call to refresh_bearer_token_from_file.

**Comment by grosser** [comment]
fixed

**Comment by cben** [comment]
Code is neat :+1:

About performance: you mentioned several options on #530 and this implements the last one:

* reload on every request (in def rest_client and def http_options) since it's pretty cheap and usually an in-memory file

So this will add several syscalls on every request.  But only when config uses a token file, which is probably mainly used in-cluster.
I don't know how to judge if that's Good Enough, but OK let's start with it and optimize later if needed.

**Pull request merged**